### PR TITLE
build: replace libdmg and genisoimage with xorriso

### PR DESCRIPTION
**Pull request by fanquake**
TLDR: Replace two tools that we build and maintain in depends, as part of our macOS deployment, with one that we don't.

Carl mentioned that [`xorriso`](https://www.gnu.org/software/xorriso/) ([Debian package](https://packages.debian.org/search?keywords=xorriso)) could replace our usage of `libdmg-hfsplus` and `genisoimage`. Looks like that's possible. Have tested gitian builds, and reproducibility looks good. Looking for Concept ACKs.

Also related to #18128, as this eliminates some dependencies. Will post thoughts on using a `.zip` for distribution there.

#### TODO:
- [ ] `xorrisofs` vs calling `xorriso -as mkisofs`
- [ ] Compression. Passing `zisofs level=9` etc.

### CODE REVIEW

**Comment by dongcarl** [comment]
Should we add `native_xorriso` to depends?

**Comment by jonasschnelli** [comment]
Nice work!
Gitian build works. Tested the DMG https://bitcointools.jonasschnelli.ch/gitian/build/10.

The only difference I observed is, that the new image is not compressed:

Master is around 12 MB (bitcoin-0.19.99-osx-unsigned-2.dmg) where this PR is around 27 (uncompressed).
```
-rw-r--r--@ 1 jonasschnelli  staff  12494037 15 Feb 21:38 bitcoin-0.19.99-osx-unsigned-2.dmg
-rw-r--r--@ 1 jonasschnelli  staff  27015168 15 Feb 21:36 bitcoin-0.19.99-osx-unsigned.dmg
```

**Comment by laanwj** [comment]
Concept ACK. Nice. Having to maintain less software as part of the build is good.

**Comment by jonasschnelli** [comment]
I'm unsure if xorriso supports HFS+ compression.
The libdmg-hfsplus code for compression looks quite extensive https://github.com/planetbeing/libdmg-hfsplus/commit/fc8a21682111a89d1700c4597780d15cae8ddb47 
and I haven't seen a similar implementation in xorrisofs code.

If it doesn't support HFS+ compression, we might want to discuss if a uncompressed disk image is acceptable (27MB instead of 12).

**Comment by laanwj** [comment]
That's a good point. I think that'd be a waste. We might first want to wait (or encourage someone to work on) adding compression there then.

**Comment by MarcoFalke** [comment]
Even with pruning enabled, the savings of 15 MB are negligible when compared to the minimum disk usage Bitcoin Core requires. So while the savings are nice, it shouldn't be a requirement?

**Comment by fanquake** [comment]
Rather than doing this, I think switching to a `.zip` is the [better solution](https://github.com/bitcoin/bitcoin/issues/18128#issuecomment-633559692), and will result in even less macOS specific code/dependencies for the project to maintain going forward. Going to close this and PR some other changes shortly.
